src/lib.rs

```diff
@@ -172,6 +172,13 @@ impl<C: ManageConnection + Send> Pool<C> {
             return self.try_get_connection().await;
         }
 
+        // We go through the max size loop here in case there are no valid
+        // connections in the pool, e.g. due to the network disconnect, so
+        // that we can immediately purge stale connections and initiate the
+        // new connections in one shot.
+        //
+        // Note that since we just returns error, in case no valid connections,
```

Comment:
I'd call out here that we return an error if we're unable to create a *new* connection after the first one is invalid

Reply:
Thanks @nate-onesignal and I've updated the comment, as I understand the logic better now than before.  Please review.